packages/@aws-cdk/pipelines/test/builds.test.ts

```diff
@@ -25,562 +26,592 @@ afterEach(() => {
-    sourceArtifact,
-    cloudAssemblyArtifact,
-    synthAction: npmYarnBuild(npmYarn)({
+behavior('%s build respects subdirectory', (suite) => {
+  suite.each(['npm', 'yarn']).legacy((npmYarn) => {
+    // WHEN
+    new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
       sourceArtifact,
       cloudAssemblyArtifact,
-      subdirectory: 'subdir',
-    }),
-  });
+      synthAction: npmYarnBuild(npmYarn)({
+        sourceArtifact,
+        cloudAssemblyArtifact,
+        subdirectory: 'subdir',
+      }),
+    });
 
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
-    Environment: {
-      Image: 'aws/codebuild/standard:5.0',
-    },
-    Source: {
-      BuildSpec: encodedJson(deepObjectLike({
-        phases: {
-          pre_build: {
-            commands: arrayWith('cd subdir'),
+    // THEN
+    expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
+      Environment: {
+        Image: 'aws/codebuild/standard:5.0',
+      },
+      Source: {
+        BuildSpec: encodedJson(deepObjectLike({
+          phases: {
+            pre_build: {
+              commands: arrayWith('cd subdir'),
+            },
           },
-        },
-        artifacts: {
-          'base-directory': 'subdir/cdk.out',
-        },
-      })),
-    },
+          artifacts: {
+            'base-directory': 'subdir/cdk.out',
+          },
+        })),
+      },
+    });
   });
 });
 
-test.each([['npm'], ['yarn']])('%s build sets UNSAFE_PERM=true', (npmYarn) => {
-  // WHEN
-  new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
-    sourceArtifact,
-    cloudAssemblyArtifact,
-    synthAction: npmYarnBuild(npmYarn)({
+behavior('%s build sets UNSAFE_PERM=true', (suite) => {
+  suite.each(['npm', 'yarn']).legacy((npmYarn) => {
+    // WHEN
+    new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
       sourceArtifact,
       cloudAssemblyArtifact,
-    }),
-  });
+      synthAction: npmYarnBuild(npmYarn)({
+        sourceArtifact,
+        cloudAssemblyArtifact,
+      }),
+    });
 
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
-    Environment: {
-      EnvironmentVariables: [
-        {
-          Name: 'NPM_CONFIG_UNSAFE_PERM',
-          Type: 'PLAINTEXT',
-          Value: 'true',
-        },
-      ],
-    },
+    // THEN
+    expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
+      Environment: {
+        EnvironmentVariables: [
+          {
+            Name: 'NPM_CONFIG_UNSAFE_PERM',
+            Type: 'PLAINTEXT',
+            Value: 'true',
+          },
+        ],
+      },
+    });
   });
 });
 
-test.each([['npm'], ['yarn']])('%s assumes no build step by default', (npmYarn) => {
-  // WHEN
-  new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
-    sourceArtifact,
-    cloudAssemblyArtifact,
-    synthAction: npmYarnBuild(npmYarn)({ sourceArtifact, cloudAssemblyArtifact }),
-  });
+behavior('%s assumes no build step by default', (suite) => {
```

Comment:
The output (as seen from IntelliJ) is not great, but it's clear enough:

![Screenshot 2021-06-22 at 15 02 48](https://user-images.githubusercontent.com/288203/122938724-ea8c7600-d36a-11eb-9d0d-bfc79a829ae3.png)

If we have some time later, we can polish this.